js/src/yoastseo-woo-plugin.js

```
@@ -1,55 +1,54 @@
-(function() {
-
-	var AssessmentResult = require( 'yoastseo/js/values/AssessmentResult' );
-
-	/**
-	 * Adds eventlistener to load the Yoast WooCommerce plugin
-	 */
-	if ( typeof YoastSEO !== 'undefined' && typeof YoastSEO.app !== 'undefined' ) {
-		new YoastWooCommercePlugin();
-	}
-	else {
-		jQuery( window ).on(
-			'YoastSEO:ready',
-			function() {
-				new YoastWooCommercePlugin();
-			}
-		);
-	}
-
+/* global YoastSEO, tinyMCE, wpseoWooL10n */
+var AssessmentResult = require( "yoastseo/js/values/AssessmentResult" );
+( function() {
 	/**
 	 * Registers Plugin and Test for Yoast WooCommerce.
+	 * @returns {void}
 	 */
 	function YoastWooCommercePlugin() {
-		YoastSEO.app.registerPlugin( 'YoastWooCommerce', { 'status': 'ready' } );
+		YoastSEO.app.registerPlugin( "YoastWooCommerce", { status: "ready" } );
 
-		YoastSEO.app.registerAssessment( 'productTitle', { getResult: this.productDescription.bind( this ) }, 'YoastWooCommerce' );
+		YoastSEO.app.registerAssessment( "productTitle", { getResult: this.productDescription.bind( this ) }, "YoastWooCommerce" );
 
 		this.addCallback();
+		this.addImageToContent();
 
-		YoastSEO.app.registerPlugin( 'YoastWooCommercePlugin', { status: 'ready' } );
+		YoastSEO.app.registerPlugin( "YoastWooCommercePlugin", { status: "ready" } );
 
 		this.registerModifications();
 
 		this.bindEvents();
 	}
 
+	/**
+	 * Adds eventlistener to load the Yoast WooCommerce plugin
+	 */
+	if( typeof YoastSEO !== "undefined" && typeof YoastSEO.app !== "undefined" ) {
+		( () => new YoastWooCommercePlugin() )();
```

Is there a specific reason for rewriting the existing code to new JS standards?

---

Its a way to not have eslint complain about 'new Foo()' instead of 'var foo = new Foo()', but I can disable this error too

---

Okay, but here https://github.com/Yoast/wpseo-woocommerce/pull/236/files#diff-dd1249ecc1aea9460170266028da2f1eL263 you didn't use the same way.

---

I didn't, but I will be suppressing the error since I think this way of dealing with it is a dirty hack.